index.js

```diff
@@ -142,6 +141,7 @@ const factory = ({
   importsAsModuleRequests = false,
   replaceInSelectors = false,
 } = {}) => async (root, rootResult) => {
+  const cache = new Map();
```

Comment:
I debated for a while what scope to put the cache in:

module scope:
Would cause issues if the files change, or the options change.

factory scope:
`postcss-loader` requires the plugin on each run, so this didn't provide much benefit. There's also the risk of people re-using the same plugin instance after files have changed.

Single process scope:
The safest, as I can assume files won't change with in a single run. In my performance testing in our large webpack build with many css files, it was only a little bit slower so I went with this.